TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/utils.py

```diff
@@ -102,3 +102,73 @@ def __call__(self, r: PreparedRequest) -> PreparedRequest:
         signer.add_auth(aws_request)
         r.headers.update(dict(aws_request.headers))
         return r
+
+
+def _map_sigv4_config(sigv4_config) -> dict:
+    sigv4_mapped = {}
+    if isinstance(sigv4_config, dict):
+        if "region" in sigv4_config:
+            sigv4_mapped["region"] = sigv4_config["region"]
+        if "service" in sigv4_config:
+            sigv4_mapped["service"] = sigv4_config["service"]
+    return {"sigv4": sigv4_mapped if sigv4_mapped else None}
+
+
+def _map_basic_auth_config(basic_config) -> dict:
+    if not isinstance(basic_config, dict):
+        raise ValueError("authConfig.basic must be a dictionary")
+    
+    if "secretName" in basic_config:
+        return {"basic_auth": {"k8s_secret_name": basic_config["secretName"]}}
+    
+    if "secretArn" in basic_config:
+        return {"basic_auth": {"user_secret_arn": basic_config["secretArn"]}}
+    
+    if "username" in basic_config and "password" in basic_config:
+        return {"basic_auth": {"username": basic_config["username"], "password": basic_config["password"]}}
+    
+    raise ValueError("authConfig.basic must contain either a secret or username/password")
+
+
+def _map_cluster_auth_from_workflow_config(auth_config) -> dict:
+    if auth_config is None or not isinstance(auth_config, dict):
+        return {"no_auth": None}
+    
+    if "sigv4" in auth_config:
+        return _map_sigv4_config(auth_config.get("sigv4"))
+    
+    if "mtls" in auth_config:
+        raise NotImplementedError("MTLS auth is not currently supported.")
```

Comment:
what will this do?  If I have a source that's otherwise setup fine and a target has mtls, will the config be valid for the source?  Is throwing really an option here?